custom_components/frigate/__init__.py

```diff
@@ -420,6 +420,7 @@ def __init__(
         config_entry: ConfigEntry,
         frigate_config: dict[str, Any],
         state_topic_config: dict[str, Any],
```

Comment:
Perhaps this would be better to take a dictionary of configs, rather than having arguments for exactly 2 configs? i.e.

`topic_configs: dict[str, dict[str, any]]`

... then you can pass in as many as you want.

Reply:
[If that's not clear, this would be kind of equivalent to passing in the variable you have named `topic_map`]

Reply:
Okay cool, I got this implemented and I like it a lot, will make it very nicely expandable in the future. Let me know if there are any adjustments needed. I left the availability topic inside the parent MQTTEntity class since I figured every entity will use that and no reason to duplicate that code